core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3809,6 +3809,29 @@ public static List distinct(List list) {
     return Collections.nCopies(numberOfElement, element);
   }
 
+  /** Support the ARRAY_EXCEPT function. */
+  public static List arrayExcept(List list1, List list2) {
+    final Set result = new LinkedHashSet<>(list1);
+    result.removeAll(list2);
+    return new ArrayList<>(result);
+  }
+
+  /** Support the ARRAY_INTERSECT function. */
+  public static List arrayIntersect(List list1, List list2) {
+    final Set result = new LinkedHashSet<>(list1);
+    result.retainAll(list2);
+    return new ArrayList<>(result);
+  }
+
+  /** Support the ARRAY_UNION function. */
+  public static List arrayUnion(List list1, List list2) {
+    final Set result =
+        new LinkedHashSet<>(Math.max((int) ((list1.size() + list2.size()) / .75f) + 1, 16));
```

Comment:
Defining some variables will make it easier to understand, why `.75f`, why 16?
`Math.max((int) ((list1.size() + list2.size()) / .75f) + 1, 16)`.

Reply:
refer to multisetUnionDistinct in SqlFunctions

Reply:
and i fixed in case of not easy to understand